packages/elements/src/internals/machines/sign-in.types.ts

```diff
@@ -1,20 +1,99 @@
-import type { EnvironmentResource, LoadedClerk } from '@clerk/types';
-import type { SnapshotFrom } from 'xstate';
-
-import type { SignInMachine } from './sign-in.machine';
+import type {
+  ClerkAPIError,
+  EmailCodeStrategy,
+  EmailLinkStrategy,
+  EnvironmentResource,
+  LoadedClerk,
+  OAuthStrategy,
+  PasswordStrategy,
+  PhoneCodeStrategy,
+  ResetPasswordEmailCodeStrategy,
+  ResetPasswordPhoneCodeStrategy,
+  SamlStrategy,
+  TicketStrategy,
+  Web3Strategy,
+} from '@clerk/types';
 
 export type FieldDetails = {
   type?: string;
   value?: string | readonly string[] | number;
-  error?: Error;
+  error?: ClerkAPIError;
 };
 
 export type WithClerk<T> = { clerk: LoadedClerkWithEnv } & T;
 export type WithClient<T> = { client: LoadedClerkWithEnv['client'] } & T;
 export type WithParams<T> = { params: T };
 
+// ====================== CLERKJS MODIFICATIONS ======================
+
 export interface LoadedClerkWithEnv extends LoadedClerk {
+  mode: 'browser' | 'server';
   __unstable__environment: EnvironmentResource;
 }
 
-export type SnapshotState = SnapshotFrom<typeof SignInMachine>;
+export interface SignInCreateOAuthParams {
```

Comment:
Better typing for `SignIn*Params`. Might make sense later to find its way into the `types` package.

Reply:
@tmilewski What's the specific improvement here over the existing types?

Reply:
Creating the separate interfaces was a nicety, but there were cases where the types weren't being appropriately narrowed. 

This was particularly seen in relation to: 

- `SignInCreateIdentifierOnlyParams` and `SignInCreateTransferOnlyParams` where I added `never` to `strategy`.

- `SignInCreateOAuthParams` and `SignInCreateSamlParams` where `identifier` is required in one but not in the other.